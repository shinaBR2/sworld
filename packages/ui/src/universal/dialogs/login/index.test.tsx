import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { LoginDialog } from '../index';

// Mock Material-UI Dialog
vi.mock('@mui/material/Dialog', () => ({
  default: ({ children, open }: { children: React.ReactNode; open: boolean }) =>
    open ? <div data-testid="dialog">{children}</div> : null,
}));

describe('LoginDialog', () => {
  it('renders correctly when open', () => {
    render(<LoginDialog />);

    // Check if dialog is rendered
    expect(screen.getByTestId('dialog')).toBeInTheDocument();

    // Check welcome text
    expect(screen.getByText('Welcome Back')).toBeInTheDocument();

    // Check button text
    expect(screen.getByText('FLOW')).toBeInTheDocument();

    // Check if Google icon is present
    expect(screen.getByTestId('GoogleIcon')).toBeInTheDocument();
  });

  it('calls onAction when login button is clicked', () => {
    const mockOnAction = vi.fn();

    render(<LoginDialog onAction={mockOnAction} />);

    const loginButton = screen.getByText('FLOW');
    fireEvent.click(loginButton);

    expect(mockOnAction).toHaveBeenCalledTimes(1);
  });

  it('does not throw when clicking button with no onAction handler', () => {
    render(<LoginDialog />);

    const loginButton = screen.getByText('FLOW');

    expect(() => fireEvent.click(loginButton)).not.toThrow();
  });

  it('renders with correct button styles', () => {
    render(<LoginDialog />);

    const button = screen.getByText('FLOW').closest('button');
    expect(button).toHaveClass('MuiButton-outlined');
    expect(button).toHaveClass('MuiButton-outlinedPrimary');
  });

  it('renders dialog content with correct MUI classes', () => {
    render(<LoginDialog />);

    const dialogContent = screen.getByTestId('dialog').querySelector('.MuiDialogContent-root');
    expect(dialogContent).toBeInTheDocument();
  });

  it('renders box with flex layout', () => {
    render(<LoginDialog />);

    const box = screen.getByTestId('dialog').querySelector('.MuiBox-root');
    expect(box).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<LoginDialog />);
    expect(container).toMatchSnapshot();
  });
});

describe('LoginDialog Integration', () => {
  it('dialog cannot be closed with escape key', () => {
    render(<LoginDialog />);

    fireEvent.keyDown(screen.getByTestId('dialog'), {
      key: 'Escape',
      code: 'Escape',
    });

    expect(screen.getByTestId('dialog')).toBeInTheDocument();
  });
});
