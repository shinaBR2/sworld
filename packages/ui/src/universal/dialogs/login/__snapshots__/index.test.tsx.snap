// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginDialog > matches snapshot 1`] = `
<div>
  <div
    data-testid="dialog"
  >
    <div
      class="MuiDialogContent-root css-1ebihfo-MuiDialogContent-root"
    >
      <div
        class="MuiBox-root css-1h93ewi"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Welcome Back
        </h5>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1poe5fq-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GoogleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
              />
            </svg>
          </span>
          FLOW
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
